### PR TITLE
fix: card error not show and cannot dismiss in report fraud page

### PR DESCRIPTION
**Pull request by dominictb**
### Fixed Issues

$ https://github.com/Expensify/App/issues/49010
PROPOSAL: https://github.com/Expensify/App/issues/49010#issuecomment-2345118755

### Tests

**Precondition**: Have a workspace with Expensify Card enabled

1. [Admin] Go to workspace settings > Expensify Card > Issue card
2. Assign a _Virtual card_ to employee with _$100 Smart Limit_
5. [Employee] Go to Settings > Wallet (might need to refresh page)
6. Verify the card shows under Assigned cards
7. [Admin] Open the card
4. Update Card limit to $0
8. [Employee] Open the card
9. Press Report virtual card fraud > Deactivate Card

Verify an error shows above Deactivate Card button

9. Close the RHP
10. Dismiss the error under the card in Assigned cards

Verify the error and RBR next to Wallet are dismissed

11. [Admin] Open the card
12. Update Card limit to $10

Verify the error below the card can be closed

13. Refresh the page

Verify the error is not there anymore

- [x] Verify that no errors appear in the JS console

### QA Steps

**Note**: See MacOS: Chrome / Safari recordings for full test steps.

**Precondition**: Have a workspace with Expensify Card enabled

1. [Admin] Go to workspace settings > Expensify Card > Issue card
2. Assign a _Virtual card_ to employee with _$100 Smart Limit_
5. [Employee] Go to Settings > Wallet (might need to refresh page)
6. Verify the card shows under Assigned cards
7. [Admin] Open the card
4. Update Card limit to $0
8. [Employee] Open the card
9. Press Report virtual card fraud > Deactivate Card

Verify an error shows above Deactivate Card button

9. Close the RHP
10. Dismiss the error under the card in Assigned cards

Verify the error and RBR next to Wallet are dismissed

11. [Admin] Open the card
12. Update Card limit to $10

Verify the error below the card can be closed

13. Refresh the page

Verify the error is not there anymore

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/user-attachments/assets/48f33efb-27e0-4b95-ab08-c9d3edea7abe

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/user-attachments/assets/643b0f37-5bed-4d0c-a479-9fc43eabc385

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/user-attachments/assets/56219682-d2ad-4ef2-9f49-f372b2a33097

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/user-attachments/assets/63b52024-57bb-415d-82f5-ba951b0e984d

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/user-attachments/assets/a81ca230-784f-4ef4-b37d-d99524d42f8b

https://github.com/user-attachments/assets/2d93a489-88ae-4303-9ae8-ab6fa9a5733a

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/user-attachments/assets/7a1fb0a1-ef43-4ef9-a506-c9877b3db074

</details>

### CODE REVIEW

**Comment by jayeshmangwani** [comment]
> I cannot reproduce the exact error message in OP. I can only reproduce Auth RequestReplacementExpensifyCard returned an error following steps in PR description.

Thanks for the thorough testing steps. I tested it by reproducing the `Auth RequestReplacementExpensifyCard` error.

**Comment by jayeshmangwani** [comment]
@dominictb Can we please fix the lint failing ? 
Even though they're not caused by our changes, we've edited a page that uses deprecated code, so can we please fix those as well? For context, see here https://expensify.slack.com/archives/C01GTK53T8Q/p1726169039084589

**Comment by jayeshmangwani** [comment]
@dominictb friendly bump for this https://github.com/Expensify/App/pull/49292#issuecomment-2358396847

**Comment by jayeshmangwani** [comment]
@dominictb Could you please merge the latest main branch and push it? The Tests are failing on the PR.

**Comment by jayeshmangwani** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/user-attachments/assets/942544ed-22c4-4351-88c9-ddf7983d299c




</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/user-attachments/assets/0a43b4f9-d155-4d41-942b-9dfd1cf02117




</details>

<details>
<summary>iOS: Native</summary>

https://github.com/user-attachments/assets/ff2f6927-c338-4593-815c-8bb32c2f0d59




</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/user-attachments/assets/aef69cb2-9f9e-48d4-8271-d111c71774f3




</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/user-attachments/assets/fd918c53-8c64-4a56-9ff0-14c9eb836cef




</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/user-attachments/assets/cdfcd6db-6183-4c85-a482-efee6ebdfcd1




</details>

File: src/pages/settings/Wallet/PaymentMethodList.tsx [hunk 1]
```diff
@@ -399,7 +409,7 @@ function PaymentMethodList({
                             icon={Expensicons.CreditCard}
                             onPress={onPress}
                             // eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing
-                            isDisabled={isLoadingPaymentMethods || isFormOffline}
+                            isDisabled={isLoadingPaymentMethods || isFormOffline || isLoadingPaymentMethodsOnyx}
```
Comment (jayeshmangwani) [inline src/pages/settings/Wallet/PaymentMethodList.tsx:412]: @dominictb Why do we need isLoadingPaymentMethodsOnyx? isLoadingPaymentMethods should do the work here, right or am i missing something?

Comment (jayeshmangwani) [reply src/pages/settings/Wallet/PaymentMethodList.tsx:412]: Thanks for the details @dominictb.

I noticed an issue on the Expensify cards page within the workspaces—pressing the close icon doesn’t clear the error message. Although it's not related to the current issue but we had fixed a similar problem here in this PR.so, Just to confirm, @dominictb @cead22  should we address it as part of this task, or create a separate bug for it?

https://github.com/user-attachments/assets/b0f078a4-e731-4bb0-99d6-897bea23a118

**Review by jayeshmangwani — APPROVED** [review]
Changes work well. I've added a comment regarding the unrelated bug here: https://github.com/Expensify/App/pull/49292#discussion_r1776631784

**Comment by jayeshmangwani** [comment]
@dominictb Please merge the main branch and resolve any conflicts. I've bumped this on [Slack](https://expensify.slack.com/archives/C02NK2DQWUX/p1727717807232729) to get someone from the internal team assigned, since @cead22 is OOO.

**Review by jayeshmangwani — APPROVED** [review]

File: src/pages/settings/Wallet/PaymentMethodList.tsx (hunk 1 above)
Comment (dangrous) [reply src/pages/settings/Wallet/PaymentMethodList.tsx:435]: Hi! Coming in here as assigned - if it's going to be a pretty similar solution let's knock it out in this PR as well. I realize it's a different page but still card-related error I guess, and if we already have done the work we need to adjust, let's do it.

**Comment by jayeshmangwani** [comment]
> Done and retested well

@dominictb Are you sure, you have tested and works well ??

**Comment by jayeshmangwani** [comment]
For me its not working

**Comment by jayeshmangwani** [comment]
Please upload a video, so that I can confirm

**Comment by jayeshmangwani** [comment]
@dangrous code looks good and is ready for your approval

**Comment by jayeshmangwani** [comment]
@dangrous Whenever you can, please check the PR. It looks good to me.

**Review by dangrous — APPROVED** [review]

**Comment by dangrous** [comment]
Thanks for your patience, I was OOO - this looks good to me!

**Pull request merged**